yt/yt/server/node/exec_node/job.cpp

```diff
@@ -987,6 +987,24 @@ bool TJob::ResourceUsageOverdrafted() const
     return TResourceHolder::GetResourceUsage().UserMemory > RequestedMemory_;
 }
 
+i64 TJob::GetOOMScoreAdjustment() const
+{
+    auto totalMemory = Bootstrap_->GetMemoryUsageTracker()->GetTotalLimit();
+    auto reservedMemory = RequestedMemory_;
+    const i64 MaxOOMScore = 1000;
+
+    if (reservedMemory > totalMemory) {
+	reservedMemory = totalMemory;
+    }
+
+    // OOM score is memory usage normalized to total memory size or cgroup memory limit.
+    // This adjustment allows to subtract score for reserved memory. The same formula is
+    // used in Kubernetes.
+    //
+    // TODO(khlebnikov): Use memory limit dedicated for user jobs cgroup.
+    return MaxOOMScore - (MaxOOMScore * reservedMemory + totalMemory - 1) / totalMemory;
```

Comment:
Can't this TODO be easily fixed by taking UserJobs limit from memory usage tracker?